felix/bpf/ut/pol_prog_test.go

```diff
@@ -2639,101 +2641,170 @@ func TestPolicyProgramsFlowLog(t *testing.T) {
-												Action: "Allow",
-												Protocol: &proto.Protocol{
-													NumberOrName: &proto.Protocol_Name{Name: "tcp"},
-												},
+						},
+						{
+							Name: "protocols",
+							Policies: []polprog.Policy{
+								{
+									Name:      "TCP allow - staged",
+									Staged:    true,
+									NoMatchID: 0x66,
+									Rules: []polprog.Rule{{
+										MatchID: 0x6,
+										Rule: &proto.Rule{
+											Action: "Allow",
+											Protocol: &proto.Protocol{
+												NumberOrName: &proto.Protocol_Name{Name: "tcp"},
 											},
-										}},
-									},
-									{
-										Name:      "UDP allow - staged",
-										Staged:    true,
-										NoMatchID: 0x1717,
-										Rules: []polprog.Rule{{
-											MatchID: 0x17,
-											Rule: &proto.Rule{
-												Action: "Allow",
-												Protocol: &proto.Protocol{
-													NumberOrName: &proto.Protocol_Name{Name: "udp"},
-												},
+										},
+									}},
+								},
+								{
+									Name:      "UDP allow - staged",
+									Staged:    true,
+									NoMatchID: 0x1717,
+									Rules: []polprog.Rule{{
+										MatchID: 0x17,
+										Rule: &proto.Rule{
+											Action: "Allow",
+											Protocol: &proto.Protocol{
+												NumberOrName: &proto.Protocol_Name{Name: "udp"},
 											},
-										}},
-									},
-									{
-										Name:  "allow all",
-										Rules: []polprog.Rule{{Rule: &proto.Rule{Action: "Allow"}, MatchID: 1}},
-									},
+										},
+									}},
+								},
+								{
+									Name:  "allow all",
+									Rules: []polprog.Rule{{Rule: &proto.Rule{Action: "Allow"}, MatchID: 1}},
 								},
 							},
 						},
 					},
-					allowedPackets: []testFlowLogCase{
-						{
-							packet:  udpPkt("10.0.0.1:31245", "10.0.0.2:80"),
-							matches: []uint64{1234, 0xdead, 0x600d, 0x7455, 0x66, 0x17, 1},
-						},
-						{
-							packet:  tcpPkt("10.0.0.2:80", "10.0.0.1:31245"),
-							matches: []uint64{1234, 0xdead, 0x600d, 0x7455, 0x6, 0x1717, 1},
-						},
-						{
-							packet:  icmpPkt("10.0.0.1", "10.0.0.2"),
-							matches: []uint64{1234, 0xdead, 0x600d, 0x7455, 0x66, 0x1717, 1},
-						},
+				},
+				allowedPackets: []testFlowLogCase{
+					{
+						packet:  udpPkt("10.0.0.1:31245", "10.0.0.2:80"),
+						matches: []uint64{1234, 0xdead, 0x600d, 0x7455, 0x66, 0x17, 1},
+					},
+					{
+						packet:  tcpPkt("10.0.0.2:80", "10.0.0.1:31245"),
+						matches: []uint64{1234, 0xdead, 0x600d, 0x7455, 0x6, 0x1717, 1},
+					},
+					{
+						packet:  icmpPkt("10.0.0.1", "10.0.0.2"),
+						matches: []uint64{1234, 0xdead, 0x600d, 0x7455, 0x66, 0x1717, 1},
 					},
 				},
 			},
-		*/
+		},
 	}
 
 	for _, test := range tests {
 		t.Run(fmt.Sprintf("name=%s", test.name), func(t *testing.T) { runTest(t, test.test) })
 	}
 }
 
+func TestPolicyProgramsWithStagedPolicy(t *testing.T) {
+	for i, p := range polProgramTests {
+		if p.ForIPv6 {
+			// XXX skip for now
```

Comment:
hmmm ipv6 is available now. Does it make sense to test it? If not, perhaps should be removed

Reply:
I enabled the test, and it's passing.